### DISCUSSION TITLE
Any recommendations for outputs with emojis?

### DISCUSSION THREAD

**Discussion by spenserblack** [charmbracelet/vhs#185] [General]
For example, with the below:
```
Output demo.gif

Require echo

Set FontSize 32
Set Width 1200
Set Height 600
Set LoopOffset 50%

Type "echo '😃Welcome to VHS!'" Sleep 500ms  Enter

Sleep 5s
```

I'm getting emojis overlapping text:
![demo](https://user-images.githubusercontent.com/8546709/205384573-b93a0c74-caf3-4d2d-85fe-0c61d9cf87d4.gif)

I'm curious if anyone has had success getting emojis to render well? What font family or other settings did you use?

**Comment by adriens** [comment]
Hi, take a look at these, I have no overlap : 

https://github.com/opt-nc/vhs#-published-tapes

**Reply by adriens** [reply]
:movie_camera:  https://youtu.be/nEMeppRgMZY

**Reply by spenserblack** [reply]
So basically, you added an extra space to pad the emojis? That's a useful trick, but it won't work in my case. The code I posted was just a minimal example, but I'm actually calling tools that have emoji output. E.g.
```
Type "my-tool"
Enter
# my-tool would print emojis
```
Since I don't have control of how that tool formats its outputs, I instead need some way to control the width of the emojis that are captured in the output.

**Reply by adriens** [reply]
> So basically, you added an extra space to pad the emojis?

Yes, that's the trick I found.

> but I'm actually calling tools that have emoji output. 

Hmmm, for that I guess, something should be done within VHS itself... ~or add extraspace on your tool :sweat_smile:~

**Reply by maaslalani** [reply]
@spenserblack Thanks for raising this discussion, this is something we'll definitely need to fix.

**Reply by spenserblack** [reply]
Oh, if it's an issue with vhs instead of my usage, I'll change the discussion category.

**Reply by maaslalani** [reply]
I think this is an issue with [xterm.js](https://xtermjs.org) because using emojis also causes problems without spaces on https://xtermjs.org

<img width="534" alt="image" src="https://user-images.githubusercontent.com/42545625/205744981-9ce80027-442e-46aa-a274-a1dca7f03d5f.png">

**Reply by spenserblack** [reply]
~~I wonder if characters like  あ or 笑 have this issue. If so, this might not be just an emoji issue, but a localization issue.~~

Looks like they don't have that issue. I think whatever spacing is used for those characters can be applied to emojis, too. Maybe this issue should be raised at xterm.js.

**Reply by liamwh** [reply]
I tried padding the emojis in my CLI but still don't get emojis to work, is there a solution/workaround for this?
![demo](https://github.com/charmbracelet/vhs/assets/23312359/f63a5bdf-2638-4426-8065-12c8d77ca81b)
